kubernetes/structures_pod.go

```diff
@@ -107,6 +108,17 @@ func flattenPodSpec(in v1.PodSpec) ([]interface{}, error) {
 	}
 
 	if len(in.Volumes) > 0 {
+		for i, volume := range in.Volumes {
+			// To avoid perpetual diff, remove the default service account token volume from PodSpec.
+			nameMatchesDefaultToken, err := regexp.MatchString("default-token-([a-z0-9]{5})", volume.Name)
```

Comment:
Ah, I just realized we are going to have to be a bit more clever than this. 

When the token get's auto-mounted it takes the name from the service account, so only if they don't supply a service account will it be called `default`, otherwise it will be the name of the service account. You can see how it does this in the admission controller code for it: [kubernetes/blob/master/plugin/pkg/admission/serviceaccount/admission.go#L46](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/serviceaccount/admission.go#L468)

So we'll have to actually pass down the serviceAccount name and make the regex match that. 😢 

We could just match on `-token-` but that creates a surprising edge case for someone to discover.

Reply:
oh, nice! That is really helpful. I hadn't thought to test with a service account specified. I'll put in a test to catch these cases and make it clever enough to handle them. Thanks!